src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4202,47 +4200,66 @@ private static BoundExpression SkipReferenceConversions(BoundExpression possibly
             var leftResultType = leftResult.Type;
             var rightResultType = rightResult.Type;
 
-            var resultType = node.OperatorResultKind switch
+            var resultTypeWithLeftState = node.OperatorResultKind switch
```

Comment:
The "Type" part of "resultType" is out-of-date.
Consider `resultTypeWithLeftState` -> `leftConvertedResult` (or re-use `leftResult`), `getLeftResultType` -> `getLeftResult` and `getResultStateWithRightType` -> `getLeftResultWithRightType`

Reply:
I don't really know what you mean. This is exactly what it purports to be: The result `Type` of the expression, with the current state of the left node.

---
In reply to: [602562035](https://github.com/dotnet/roslyn/pull/52009#discussion_r602562035) [](ancestors = 602562035)

Reply:
I think the mismatch "with" versus "and". `resultWithLeftState` would make sense, and `resultTypeAndLeftState` would also work but seems worse.
Anyways, if either `leftResult` or `leftConvertedResult` are appropriate, that's what I'd use.

Reply:
I don't think any of these suggestions read better, tbh.

* `resultWithLeftState` - What result?
* `resultTypeAndLeftState` - That implies a tuple to me. I think `With` sounds better.
* The other two are just wrong. They're not the left result, they're the resulting type of the expression, _with_ the resulting state of the left side.

---
In reply to: [602577268](https://github.com/dotnet/roslyn/pull/52009#discussion_r602577268) [](ancestors = 602577268)

Reply:
After offline discussion, I've turned this into a tuple return and split them up.

---
In reply to: [602580327](https://github.com/dotnet/roslyn/pull/52009#discussion_r602580327) [](ancestors = 602580327,602577268)